src/hyp_map.rs

```diff
@@ -224,17 +291,23 @@ impl HypMap {
             .unwrap();
         let sv48: FirstStagePageTable<Sv48> =
             FirstStagePageTable::new(root_page).expect("creating first sv48");
-
         // Map supervisor regions
         for r in &self.supervisor_regions {
             r.map(&sv48, &mut || {
                 hyp_mem.take_pages_for_hyp_state(1).into_iter().next()
             });
         }
-        // Map user regions.
-        for r in &self.user_regions {
-            r.map(&sv48, hyp_mem);
+        // Map umode regions and create umode reset areas.
+        let mut umode_reset = ArrayVec::<UmodeResetArea, MAX_HYPMAP_UMODE_REGIONS>::new();
```

Comment:
can ArrayVec::<UmodeResetArea, MAX_HYPMAP_UMODE_REGIONS> and the non-umode version be type aliases?

Reply:
As a matter of personal preference, I prefer to expose at least a simple combination of basic types, rather than having to look it up all the time. It's verbose, but such is rust and anything camel case.

Not dying on the hill of a typing fight, though.

Reply:
FWIW, I think the generic types for `ArrayVec` will get inferred here, so there's no need to spell them out explicitly. i.e. `ArrayVec::new()` should just work.

Reply:
In the end, had to create a type for the vec in hypmap because they became so long they had to be indented.

Here I used `ArrayVec::new()`.